sources/api/apiserver/src/server/ephemeral_storage.rs

```diff
@@ -0,0 +1,379 @@
+                mdadm_create(
+                    OsString::from(NAME),
+                    disks.iter().map(OsString::from).collect(),
+                )?;
+            }
+            // can't lookup the array until it's created
+            resolve_array_by_id()?
+        }
+    };
+
+    let fs = fs.unwrap_or(Filesystem::XFS);
+    if !is_formatted(&device_name, &fs)? {
+        info!("formatting {:?} as {}", device_name, fs);
+        format_device(&device_name, &fs)?;
+    } else {
+        info!(
+            "{:?} is already formatted as {}, skipping format",
+            device_name, fs
+        );
+    }
+
+    Ok(())
+}
+
+/// binds the specified directories to the pre-configured array, creating those directories if
+/// they do not exist.
+pub fn bind(variant: &str, dirs: Vec<String>) -> Result<()> {
+    // handle the no local instance storage case
+    if ephemeral_devices()?.is_empty() {
+        info!("no ephemeral disks found, skipping ephemeral storage binding");
+        return Ok(());
+    }
+
+    let device_name = resolve_device_by_label()?;
+    let mount_point = format!("/mnt/{}", NAME);
+    let mount_point = Path::new(&mount_point);
+    let allowed_dirs = allowed_bind_dirs(variant);
+    for dir in &dirs {
+        ensure!(
+            allowed_dirs.contains(dir.as_str()),
+            error::InvalidParameterSnafu {
+                parameter: dir,
+                reason: "specified bind directory not in allow list",
+            }
+        )
+    }
+    std::fs::create_dir_all(mount_point).context(error::MkdirSnafu {})?;
+
+    info!("mounting {:?} as {:?}", device_name, mount_point);
+    ensure!(
+        Command::new(MOUNT)
+            .args([&device_name, mount_point.as_os_str()])
+            .status()
+            .context(error::MountArrayFailureSnafu {})?
+            .success(),
+        error::MountArrayFailureStatusSnafu {}
+    );
+
+    for dir in dirs {
+        // construct a directory name (E.g. /var/lib/kubelet => _var_lib_kubelet) that will be
+        // unique between the binding targets
+        let directory_name = OsString::from(dir.replace('/', "_"));
+        let mount_destination = mount_point.join(&directory_name);
+        let source_dir = OsString::from(&dir);
+
+        // we may run before the directories we are binding exist, so create them
+        std::fs::create_dir_all(&dir).context(error::MkdirSnafu {})?;
+        std::fs::create_dir_all(&mount_destination).context(error::MkdirSnafu {})?;
+
+        // call the equivalent of
+        // mount --rbind /mnt/ephemeral/_var_lib_kubelet /var/lib/kubelet
+        info!("binding {:?} to {:?}", source_dir, mount_destination);
+        ensure!(
+            Command::new(MOUNT)
+                .args([
+                    OsStr::new("--rbind"),
+                    mount_destination.as_ref(),
+                    &source_dir
+                ])
+                .status()
```

Comment:
If we're binding to multiple targets, we may error out in the second or Nth mount, and if the command were attempted again, we'd end up repeatedly binding the first mount. It might be good to check whether this directory is already a bind mount from the ephemeral storage, via parsing `findmnt -n -o TARGET,FSROOT <dir>` output. And skip it if so.